sickbeard/server/web/core/base.py

```diff
@@ -0,0 +1,471 @@
+        use_strict = sickbeard.BRANCH and sickbeard.BRANCH != 'master'
+        mako_lookup = TemplateLookup(directories=[mako_path],
+                                     module_directory=mako_cache,
+                                     #  format_exceptions=True,
+                                     strict_undefined=use_strict,
+                                     filesystem_checks=True)
+    return mako_lookup
+
+
+class PageTemplate(MakoTemplate):
+    """
+    Mako page template
+    """
+    def __init__(self, rh, filename):
+        lookup = get_lookup()
+        self.template = lookup.get_template(filename)
+
+        self.arguments = {
+            'srRoot': sickbeard.WEB_ROOT,
+            'sbHttpPort': sickbeard.WEB_PORT,
+            'sbHttpsPort': sickbeard.WEB_PORT,
+            'sbHttpsEnabled': sickbeard.ENABLE_HTTPS,
+            'sbHandleReverseProxy': sickbeard.HANDLE_REVERSE_PROXY,
+            'sbThemeName': sickbeard.THEME_NAME,
+            'sbDefaultPage': sickbeard.DEFAULT_PAGE,
+            'loggedIn': rh.get_current_user(),
+            'sbStartTime': rh.startTime,
+            'numErrors': len(classes.ErrorViewer.errors),
+            'numWarnings': len(classes.WarningViewer.errors),
+            'sbPID': str(sickbeard.PID),
+                          'title': 'FixME', 'header': 'FixME', 'topmenu': 'FixME', 'submenu': [], 'controller': 'FixME',
+                          'action': 'FixME', 'show': UNDEFINED, 'newsBadge': '', 'toolsBadge': '',
+                          'toolsBadgeClass': ''}
+
+        if rh.request.headers['Host'][0] == '[':
+            self.arguments['sbHost'] = re.match(r'^\[.*\]', rh.request.headers['Host'], re.X | re.M | re.S).group(0)
+        else:
+            self.arguments['sbHost'] = re.match(r'^[^:]+', rh.request.headers['Host'], re.X | re.M | re.S).group(0)
+        if 'X-Forwarded-Host' in rh.request.headers:
+            self.arguments['sbHost'] = rh.request.headers['X-Forwarded-Host']
+        if 'X-Forwarded-Port' in rh.request.headers:
+            sbHttpPort = rh.request.headers['X-Forwarded-Port']
+            self.arguments['sbHttpsPort'] = sbHttpPort
+        if 'X-Forwarded-Proto' in rh.request.headers:
+            self.arguments['sbHttpsEnabled'] = True if rh.request.headers['X-Forwarded-Proto'] == 'https' else False
+
+        error_count = len(classes.ErrorViewer.errors)
+        warning_count = len(classes.WarningViewer.errors)
+
+        if sickbeard.NEWS_UNREAD:
+            self.arguments['newsBadge'] = ' <span class="badge">{news}</span>'.format(news=sickbeard.NEWS_UNREAD)
+
+        numCombined = error_count + warning_count + sickbeard.NEWS_UNREAD
+        if numCombined:
+            if error_count:
+                self.arguments['toolsBadgeClass'] = ' btn-danger'
+            elif warning_count:
+                self.arguments['toolsBadgeClass'] = ' btn-warning'
+            self.arguments['toolsBadge'] = ' <span class="badge{type}">{number}</span>'.format(
```

Comment:
When the line length isn't excessive I prefer 1.
2. can't be used here since the format isn't inside a function (or else a `\` would be required to wrap to the next line).
In some cases 3 leads to a wrapped indent that doesn't neatly line up with the previous line.

I generally prefer them in the order listed, myself.